pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -1347,113 +1348,64 @@ class Timedelta(_Timedelta):
     __rsub__ = _binary_op_method_timedeltalike(lambda x, y: y - x, '__rsub__')
 
     def __mul__(self, other):
-        if hasattr(other, '_typ'):
-            # Series, DataFrame, ...
-            if other._typ == 'dateoffset' and hasattr(other, 'delta'):
-                # Tick offset; this op will raise TypeError
-                return other.delta * self
```

Comment:
This part is covered by `rmul` of DateOffset?

Reply:
It will go through `DateOffset.__rmul__` which will raise TypeError (correctly), yes.